### PR TITLE
#157958385 Completes user login frontend

### PR DESCRIPTION
**Pull request by chukwuemekachm**
#### What does this PR do?
- Completes user login frontend client
#### Description of Task to be completed?
- Add signin.html
- Add login method to auth.js that POSTs user info to API
#### How should this be manually tested?
- Clone the repo
- Startup command line
- cd to project root
- run ``` npm install ```
- run ``` npm run start:dev ```
- Open your browser and navigate to `localhost:3000/client`
#### What are the relevant pivotal tracker stories?
- #157958385
#### Screenshots (if appropriate)

screenshot 197

#### Questions:
- None

### CODE REVIEW

File: public/js/auth.js [hunk 1]
```diff
@@ -1,32 +1,33 @@
...
-    .then((res) => {
-      if (res.code === 403) window.location.replace('user.html');
-      if (res.code === 200) window.location.replace('admin.html');
-    }).catch((err) => {
-      displayAlert(`Welcome ${err.message}, your login failed`, 2);
-    });
+  .then(res => res.json())
+  .then((res) => {
+    if (res.code === 403) window.location.replace('user.html');
+    if (res.code === 200) window.location.replace('admin.html');
```
Comment (benfluleck) [inline public/js/auth.js:19]: Do you mean Status code?

File: public/js/auth.js [hunk 2]
```diff
@@ -55,9 +56,41 @@ if (formSignup) {
...
+    const email = document.getElementById('email').value;
+    const password = document.getElementById('password').value;
+
+    if (email && password) {
+      const body = { email, password };
+      fetch(`${baseUrl}/auth/login`, {
+        method: 'POST',
+        body: JSON.stringify(body),
+        headers: { 'Content-Type': 'application/json' },
+        mode: 'cors',
```
Comment (benfluleck) [inline public/js/auth.js:78]: why is course being enabled here ?

File: public/js/auth.js [hunk 3]
```diff
@@ -55,9 +56,41 @@ if (formSignup) {
...
-        .catch((err) => { displayAlert(err.message, 1); });
+      .catch((err) => { displayAlert(err.message, 1); });
     } else {
       displayAlert('Please fill in the form', 2);
     }
   });
 }
+
+if (formLogin) {
+  formLogin.addEventListener('submit', (e) => {
```
Comment (benfluleck) [inline public/js/auth.js:67]: Go and research on removeEventListener after adding

File: public/js/auth.js [hunk 4]
```diff
@@ -55,9 +56,41 @@ if (formSignup) {
...
+    e.preventDefault();
```
Comment (benfluleck) [inline public/js/auth.js:68]: What is e?

File: public/signin.html [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    </div>
+  </div>
+  <div class="ch-footer ch-card">
+    <h5>Copyright of &copy;2018 Maintenance Tracker by Chima Chukwuemeka</h5>
+  </div>
+</body>
+<script src="js/script.js"></script>
+<script src="js/auth.js"></script>
+
+</html>
```
Comment (benfluleck) [inline public/signin.html:49]: Why is there no new line there ??

File: public/js/auth.js [hunk 6]
```diff
@@ -55,9 +56,41 @@ if (formSignup) {
...
+        displayAlert(res.message, 2);
+      })
+      .catch((err) => {
+        displayAlert(`${err.message}, your login failed`, 2);
+      });
+    } else {
+      displayAlert('Please fill in the form', 2);
+    }
+  });
+}
```
Comment (benfluleck) [inline public/js/auth.js:96]: ???

File: public/js/auth.js (hunk 1 above)
Comment (chukwuemekachm) [reply public/js/auth.js:19]: Yes, it is.
My JSend response object has a code property, so I'm using the value of the code property to make decisions after resolving the promise.

File: public/js/auth.js (hunk 4 above)
Comment (chukwuemekachm) [reply public/js/auth.js:68]: The submit event

File: public/js/auth.js (hunk 3 above)
Comment (chukwuemekachm) [reply public/js/auth.js:67]: Ok
Thanks

File: public/js/auth.js (hunk 6 above)
Comment (chukwuemekachm) [reply public/js/auth.js:96]: To check form fields.
Thanks, I get it now, it's not necessary

File: public/js/auth.js (hunk 2 above)
Comment (chukwuemekachm) [reply public/js/auth.js:78]: I used it to test for Internet Explorer.
But forgot to remove it

**Pull request merged**